quilt/data.py

```diff
@@ -79,6 +72,16 @@ def __dir__(self):
         # https://mail.python.org/pipermail/python-ideas/2011-May/010321.html
         return sorted(set((dir(type(self)) + list(self.__dict__) + self._keys())))
 
+    def _is_group(self):
+        return isinstance(self._node, GroupNode)
+
+    def _is_df(self):
+        return not isinstance(self._node, GroupNode)
+
+    def _df(self):
```

Comment:
`_df()` is OK for now. suggest `_dfkeys()` and `_groupkeys()` for the other methods; will need to `README.md` with new APIs